stl/inc/ranges

```diff
@@ -1913,6 +1913,49 @@ namespace ranges {
         };
 
         inline constexpr _Drop_fn drop;
+
+        // VARIABLE views::counted
+        class _Counted_fn {
+        private:
+            enum class _St { _Span, _Subrange, _Subrange_counted };
+
+            template <class _It>
+            _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
+                _STL_INTERNAL_STATIC_ASSERT(input_or_output_iterator<_It>);
+                if constexpr (contiguous_iterator<_It>) {
+                    return {_St::_Span, noexcept(span{_STD to_address(_STD declval<_It>()), iter_difference_t<_It>{}})};
+                } else if constexpr (random_access_iterator<_It>) {
+                    return {_St::_Subrange,
+                        noexcept(subrange{_STD declval<_It>(), _STD declval<_It>() + iter_difference_t<_It>{}})};
+                } else {
+                    return {_St::_Subrange_counted,
+                        noexcept(subrange{
+                            counted_iterator{_STD declval<_It>(), iter_difference_t<_It>{}}, default_sentinel})};
+                }
+            }
+
+            template <class _It>
+            static constexpr _Choice_t<_St> _Choice = _Choose<_It>();
+
+        public:
+            // clang-format off
+            template <class _It>
+                requires input_or_output_iterator<decay_t<_It>>
+            _NODISCARD constexpr auto operator()(_It&& _First, const iter_difference_t<decay_t<_It>> _Count) const
+                noexcept(_Choice<decay_t<_It>>._No_throw) {
+                // clang-format on
+                _STL_ASSERT(_Count >= 0, "The size passed to views::counted must be non-negative");
+                if constexpr (_Choice<decay_t<_It>>._Strategy == _St::_Span) {
+                    return span{_STD to_address(_STD forward<_It>(_First)), static_cast<size_t>(_Count)};
+                } else if constexpr (_Choice<decay_t<_It>>._Strategy == _St::_Subrange) {
+                    return subrange{_First, _First + _Count};
+                } else if constexpr (_Choice<decay_t<_It>>._Strategy == _St::_Subrange_counted) {

```

Comment:
Suggestion: This repeats `_Choice<decay_t<_It>>._Strategy` 3 times. In the Spaceship CPOs, I used `constexpr _St _Strat = MEOW` to make this more readable.

Reply:
Yeah, I was thinking about that one too :+1:

Reply:
I'll validate and push a change so we can move this to Ready To Meow.

Reply:
Actually, I will file an issue for a global cleanup. There are lots of occurrences in `<ranges>` and `<xutility>` that should be changed consistently.

Reply:
Filed #1419.